### PR TITLE
Fixes tiling of wallpapers after https://github.com/lxde/pcmanfm-qt/c…

### PR DESCRIPTION
**Pull request by tsujan**
Fixes tiling of wallpapers after https://github.com/lxde/pcmanfm-qt/commit/de24628202b2789d8c11954669a5b86f35b2c346.

### CODE REVIEW

File: pcmanfm/desktopwindow.cpp [hunk 1]
```diff
@@ -534,7 +534,10 @@ void DesktopWindow::paintBackground(QPaintEvent *event) {
     painter.fillRect(event->rect(), QBrush(bgColor_));
   }
   else {
-    painter.drawPixmap(event->rect(), wallpaperPixmap_, event->rect());
+    if(wallpaperMode_ == WallpaperTile)
```
Comment (palinek) [inline pcmanfm/desktopwindow.cpp]: Shouldn't be handled all cases here?
```c++
  enum WallpaperMode {
    WallpaperNone,
    WallpaperStretch,
    WallpaperFit,
    WallpaperCenter,
    WallpaperTile,
    WallpaperZoom
  };
```

Comment (tsujan) [inline pcmanfm/desktopwindow.cpp]: What do you mean by that? After @PCMan's workaround, tiling should be done by drawTiledPixmap().

**Comment by tsujan** [comment]
@pmattern, if you have dual or hdpi monitor, could you please test this?

File: pcmanfm/desktopwindow.cpp (hunk 1 above)
Comment (palinek) [inline pcmanfm/desktopwindow.cpp]: Sorry... it's done [here](https://github.com/lxde/pcmanfm-qt/blob/master/pcmanfm/desktopwindow.cpp#L297)

Comment (tsujan) [inline pcmanfm/desktopwindow.cpp]: It's OK.

**Comment by palinek** [comment]
I've made a test... and with this patch, the tiling has the same "bad" symptoms as described in the original issue lxde/lxqt#1027. We probably need to render the tilling manually -> with this [commit](https://github.com/lxde/pcmanfm-qt/commit/bedd02ca8141), the tilling is OK for me.

**Comment by tsujan** [comment]
@palinek 
Thanks for the test! Your approach seems good but please read my comment at https://github.com/lxde/pcmanfm-qt/commit/bedd02ca8141#commitcomment-20661937

**Comment by tsujan** [comment]
There may still be a problem and that's HDPI. `QPainter::drawTiledPixmap()` is not equivalent to drawing a pixmap several times.

**Comment by palinek** [comment]
> QPainter::drawTiledPixmap() is not equivalent to drawing a pixmap several times.

From the doc -> http://doc.qt.io/qt-5/qpainter.html#drawTiledPixmap
"Calling drawTiledPixmap() is similar to calling drawPixmap() several times to fill (tile) an area with a pixmap, but is potentially much more efficient depending on the underlying window system."

**Comment by tsujan** [comment]
Yes, I wanted to quote it too. I mean this part: "is potentially *much more efficient* depending on the underlying window system."

Let me first try your solution with small images and compare it to drawTiledPixmap()!

**Comment by tsujan** [comment]
Superseded by https://github.com/lxde/pcmanfm-qt/pull/448

**Pull request closed without merging**